### ISSUE TITLE
Apple M2 issue when trying to install carl-storm: ginac and cln not found

### ISSUE THREAD

**Issue by xwang2775** [moves-rwth/carl-storm#42]
Hello,

When I try to install carl-storm like this:
```
$ git clone https://github.com/moves-rwth/carl-storm
$ cd carl-storm
$ mkdir build
$ cd build
$ cmake ..
$ make lib_carl
```

there is error saying that ginac and cln is not found, but I'm sure I have installed them and set the correct environment variables
```
-- The CXX compiler identification is AppleClang 14.0.3.14030022
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Using version 14.28.
-- Version: carl 14.28
-- Detected that target system uses Apple Silicon.
CMake Warning at CMakeLists.txt:107 (message):
  Compiling natively on Apple Silicon is experimental.  Please report issues
  to support@stormchecker.org.  For more information visit
  https://www.stormchecker.org/documentation/obtain-storm/apple-silicon.html


-- Using clang 14.0.3.14030022 on Apple
-- CXX Flags:  -std=c++1y -stdlib=libc++ -pthread
-- CXX Debug Flags: -g -O1
-- CXX Release Flags: -O3 -DNDEBUG -O3 -fomit-frame-pointer -funroll-loops
-- Build type: RELEASE
-- Building static: no
-- cotire is disabled
-- GMP / GMPXX 6.3.0 was found at /opt/homebrew/include and /opt/homebrew/opt/gmp/lib/libgmp.dylib
CMake Warning (dev) at cmake/carlmacros.cmake:149 (find_package):
  Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
  --help-policy CMP0167" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

Call Stack (most recent call first):
  resources/resources.cmake:70 (load_library)
  CMakeLists.txt:168 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at cmake/carlmacros.cmake:170 (find_package):
  Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
  --help-policy CMP0167" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

Call Stack (most recent call first):
  resources/resources.cmake:70 (load_library)
  CMakeLists.txt:168 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Boost 1.85.0 was found at /opt/homebrew/include and Boost_filesystem_SHARED;Boost_system_SHARED;Boost_program_options_SHARED;Boost_regex_SHARED;Boost_unit_test_framework_SHARED
-- Eigen3 3.4.0 was found at /opt/homebrew/include/eigen3
-- Bliss is disabled
-- CLN 1.3.7 was found at /opt/homebrew/include and /opt/homebrew/lib/libcln.dylib
-- CoCoA is disabled
-- GiNaC 1.8.7 was found at /opt/homebrew/include/ginac and /opt/homebrew/lib/libginac.dylib
-- GTest 1.8.0 was found at /Users/wangxinyu/RAG/RAG-with-Verification/carl-storm/build/resources/src/GTest-EP/googletest/include and /Users/wangxinyu/RAG/RAG-with-Verification/carl-storm/build/resources/src/GTest-EP-build/googlemock/gtest/libgtest_main.a
-- Doxygen was not found.
-- Did not find clang-tidy, target tidy is disabled.
-- Did not find clang-format, target format is disabled.
-- Registered with cmake
-- Configuring done (0.6s)
-- Generating done (0.1s)
-- Build files have been written to: /Users/wangxinyu/RAG/RAG-with-Verification/carl-storm/build
(env) (/Users/wangxinyu/RAG/llm_agent/RAG-with-Verification/rag) wangxinyu@wangxinyu build % make lib_carl
[  0%] Built target resources
[  4%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/converter/CoCoAAdaptor.cpp.o
[  4%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/converter/OldGinacConverter.cpp.o
[  9%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/core/Monomial.cpp.o
[ 13%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/core/MonomialPool.cpp.o
[ 13%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/core/Variable.cpp.o
[ 18%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/core/VariablePool.cpp.o
[ 22%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/core/logging.cpp.o
[ 22%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/core/rootfinder/EigenWrapper.cpp.o
[ 27%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/formula/Sort.cpp.o
[ 31%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/formula/SortManager.cpp.o
[ 31%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/formula/bitvector/BVConstraint.cpp.o
[ 36%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/formula/bitvector/BVConstraintPool.cpp.o
[ 36%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/formula/bitvector/BVTerm.cpp.o
[ 40%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/formula/bitvector/BVTermPool.cpp.o
[ 45%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/formula/model/uninterpreted/SortValue.cpp.o
[ 45%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/formula/model/uninterpreted/SortValueManager.cpp.o
[ 50%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/formula/model/uninterpreted/UFModel.cpp.o
[ 54%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/formula/parser/OPBImporter.cpp.o
In file included from /Users/wangxinyu/RAG/RAG-with-Verification/carl-storm/src/carl/formula/parser/OPBImporter.cpp:1:
In file included from /Users/wangxinyu/RAG/RAG-with-Verification/carl-storm/src/carl/formula/parser/OPBImporter.h:5:
In file included from /Users/wangxinyu/RAG/RAG-with-Verification/carl-storm/src/carl/formula/parser/../Formula.h:1286:
/Users/wangxinyu/RAG/RAG-with-Verification/carl-storm/src/carl/formula/parser/../Formula.tpp:107:21: warning: 'pbConstraint' is deprecated: PBConstraints are normal constraints with Boolean variables now. [-Wdeprecated-declarations]
                                        for (auto var: pbConstraint().gatherVariables()) {
                                                       ^
/Users/wangxinyu/RAG/RAG-with-Verification/carl-storm/src/carl/formula/parser/../Formula.h:599:15: note: 'pbConstraint' has been explicitly marked deprecated here
            [[deprecated("PBConstraints are normal constraints with Boolean variables now.")]]
              ^
1 warning generated.
[ 54%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/formula/uninterpreted/UEquality.cpp.o
[ 59%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/formula/uninterpreted/UFInstance.cpp.o
[ 63%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/formula/uninterpreted/UFInstanceManager.cpp.o
[ 63%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/formula/uninterpreted/UFManager.cpp.o
[ 68%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/formula/uninterpreted/UninterpretedFunction.cpp.o
[ 68%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/groebner/gb-buchberger/BuchbergerStats.cpp.o
[ 72%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/numbers/adaption_cln/operations.cpp.o
[ 77%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/numbers/adaption_float/mpfr_float.cpp.o
[ 77%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/numbers/adaption_gmpxx/operations.cpp.o
[ 81%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/numbers/number/NumberClI.cpp.o
[ 86%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/numbers/number/NumberClRA.cpp.o
[ 86%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/numbers/number/NumberMpq.cpp.o
[ 90%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/numbers/number/NumberMpz.cpp.o
[ 95%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/util/BitVector.cpp.o
[ 95%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/util/CMakeOptions.cpp.o
[100%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/util/debug.cpp.o
[100%] Linking CXX shared library ../../libcarl.dylib
ld: warning: directory not found for option '-L$/opt/homebrew/opt/ginac/lib'
ld: warning: directory not found for option '-L$/opt/homebrew/opt/ginac/lib'
ld: warning: directory not found for option '-L$/opt/homebrew/opt/ginac/lib'
ld: warning: directory not found for option '-L$/opt/homebrew/opt/cln/lib'
ld: warning: directory not found for option '-L$/opt/homebrew/opt/cln/lib'
[100%] Built target lib_carl
```

This also caused further issues when I try to install pycarl (which failed for this reason) :
```
[ 60%] Building CXX object CMakeFiles/core.dir/src/mod_core.cpp.o
[ 80%] Building CXX object CMakeFiles/core.dir/src/core/variable.cpp.o
In file included from /Users/wangxinyu/RAG/RAG-with-Verification/pycarl/src/core/monomial.cpp:3:
In file included from In file included from /Users/wangxinyu/RAG/RAG-with-Verification/pycarl/src/types.h/Users/wangxinyu/RAG/RAG-with-Verification/pycarl/src/core/variable.cpp::7:
In file included from 5/Users/wangxinyu/RAG/RAG-with-Verification/pycarl/src/types.h:5:
In file included from /Users/wangxinyu/RAG/RAG-with-Verification/carl-storm/src/carl/core/MultivariatePolynomial.hIn file included from :1376/Users/wangxinyu/RAG/RAG-with-Verification/pycarl/src/mod_core.cpp:
:In file included from 6/Users/wangxinyu/RAG/RAG-with-Verification/carl-storm/src/carl/core/MultivariateGCD.h:
:In file included from In file included from /Users/wangxinyu/RAG/RAG-with-Verification/pycarl/src/typed_core/interval.h/Users/wangxinyu/RAG/RAG-with-Verification/pycarl/src/core/bound_type.cpp14::
3In file included from :
In file included from /Users/wangxinyu/RAG/RAG-with-Verification/pycarl/src/types.h/Users/wangxinyu/RAG/RAG-with-Verification/carl-storm/src/carl/core/../converter/OldGinacConverter.h::
:5:
In file included from In file included from /Users/wangxinyu/RAG/RAG-with-Verification/carl-storm/src/carl/core/MultivariatePolynomial.h/Users/wangxinyu/RAG/RAG-with-Verification/carl-storm/src/carl/core/MultivariatePolynomial.h:::137618:
1376:
/Users/wangxinyu/RAG/RAG-with-Verification/carl-storm/src/carl/core/polynomialfunctions/../../converter/ginac.hIn file included from :6:10/Users/wangxinyu/RAG/RAG-with-Verification/carl-storm/src/carl/core/MultivariateGCD.h:: 14:
In file included from /Users/wangxinyu/RAG/RAG-with-Verification/carl-storm/src/carl/core/../converter/OldGinacConverter.h4::
18In file included from :
/Users/wangxinyu/RAG/RAG-with-Verification/pycarl/src/types.h:5:
In file included from /Users/wangxinyu/RAG/RAG-with-Verification/carl-storm/src/carl/core/MultivariatePolynomial.h:1376:
In file included from fatal error: :
/Users/wangxinyu/RAG/RAG-with-Verification/carl-storm'ginac/ginac.h' file not found
/src/carl/core/MultivariateGCD.hIn file included from /Users/wangxinyu/RAG/RAG-with-Verification/carl-storm/src/carl/core/polynomialfunctions/../../converter/ginac.h:14#include <ginac/ginac.h>:
/Users/wangxinyu/RAG/RAG-with-Verification/carl-storm/src/carl/core/MultivariateGCD.h:
6In file included from /Users/wangxinyu/RAG/RAG-with-Verification/carl-storm/src/carl/core/../converter/OldGinacConverter.h         ^~~~~~~~~~~~~~~:
18:
/Users/wangxinyu/RAG/RAG-with-Verification/carl-storm/src/carl/core/polynomialfunctions/../../converter/ginac.h:6::1410:
:In file included from  /Users/wangxinyu/RAG/RAG-with-Verification/carl-storm/src/carl/core/../converter/OldGinacConverter.h:18fatal error: :
/Users/wangxinyu/RAG/RAG-with-Verification/carl-storm/src/carl/core/polynomialfunctions/../../converter/ginac.h'ginac/ginac.h' file not found:6
:10: fatal error: #include <ginac/ginac.h>
'ginac/ginac.h' file not found:
        ^~~~~~~~~~~~~~~
#include <ginac/ginac.h>
        ^~~~~~~~~~~~~~~
10: fatal error: 'ginac/ginac.h' file not found
#include <ginac/ginac.h>
        ^~~~~~~~~~~~~~~
1 error generated.
1 error generated.
1 error generated.
1 error generated.
make[3]: *** [CMakeFiles/core.dir/src/core/bound_type.cpp.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: *** [CMakeFiles/core.dir/src/mod_core.cpp.o] Error 1
make[3]: *** [CMakeFiles/core.dir/src/core/variable.cpp.o] Error 1
make[3]: *** [CMakeFiles/core.dir/src/core/monomial.cpp.o] Error 1
make[2]: *** [CMakeFiles/core.dir/all] Error 2
make[1]: *** [CMakeFiles/core.dir/rule] Error 2
make: *** [core] Error 2
Traceback (most recent call last):
 File "/Users/wangxinyu/RAG/RAG-with-Verification/pycarl/setup.py", line 182, in <module>
   setup(
 File "/Users/wangxinyu/RAG/RAG-with-Verification/storm/build/env/lib/python3.11/site-packages/setuptools/__init__.py", line 103, in setup
   return distutils.core.setup(**attrs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 File "/Users/wangxinyu/RAG/RAG-with-Verification/storm/build/env/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 184, in setup
   return run_commands(dist)
          ^^^^^^^^^^^^^^^^^^
 File "/Users/wangxinyu/RAG/RAG-with-Verification/storm/build/env/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 200, in run_commands
   dist.run_commands()
 File "/Users/wangxinyu/RAG/RAG-with-Verification/storm/build/env/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
   self.run_command(cmd)
 File "/Users/wangxinyu/RAG/RAG-with-Verification/storm/build/env/lib/python3.11/site-packages/setuptools/dist.py", line 976, in run_command
   super().run_command(command)
 File "/Users/wangxinyu/RAG/RAG-with-Verification/storm/build/env/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
   cmd_obj.run()
 File "/Users/wangxinyu/RAG/RAG-with-Verification/storm/build/env/lib/python3.11/site-packages/setuptools/command/develop.py", line 36, in run
   self.install_for_development()
 File "/Users/wangxinyu/RAG/RAG-with-Verification/storm/build/env/lib/python3.11/site-packages/setuptools/command/develop.py", line 113, in install_for_development
   self.run_command('build_ext')
 File "/Users/wangxinyu/RAG/RAG-with-Verification/storm/build/env/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
   self.distribution.run_command(command)
 File "/Users/wangxinyu/RAG/RAG-with-Verification/storm/build/env/lib/python3.11/site-packages/setuptools/dist.py", line 976, in run_command
   super().run_command(command)
 File "/Users/wangxinyu/RAG/RAG-with-Verification/storm/build/env/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
   cmd_obj.run()
 File "/Users/wangxinyu/RAG/RAG-with-Verification/pycarl/setup.py", line 154, in run
   subprocess.check_call(["cmake", "--build", ".", "--target", ext.name] + build_args, cwd=self.build_temp)
 File "/Users/wangxinyu/RAG/llm_agent/RAG-with-Verification/rag/lib/python3.11/subprocess.py", line 413, in check_call
   raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '--build', '.', '--target', 'core', '--config', 'Release', '--', '-j8']' returned non-zero exit status 2.
(env) (/Users/wangxinyu/RAG/llm_agent/RAG-with-Verification/rag) wangxinyu@wangxinyu pycarl %
```

**Comment by volkm** [comment]
CLN and Ginac are found during the configuration, see the following lines in the output:
```
-- CLN 1.3.7 was found at /opt/homebrew/include and /opt/homebrew/lib/libcln.dylib
-- CoCoA is disabled
-- GiNaC 1.8.7 was found at /opt/homebrew/include/ginac and /opt/homebrew/lib/libginac.dylib
```
The linking steps uses different locations though:
```
ld: warning: directory not found for option '-L$/opt/homebrew/opt/ginac/lib'
ld: warning: directory not found for option '-L$/opt/homebrew/opt/cln/lib'
```

Can you try uninstalling cln and ginac via Homebrew, making sure that the directories `opt/homebrew/opt/ginac` and `opt/homebrew/opt/cln` are removed, and then reinstall cln and ginac?

**Comment by xwang2775** [comment]
Thanks for your prompt replies! I have successfully installed carl-storm and built lib_carl,

> _/Users/wangxinyu/RAG/RAG-with-Verification/carl-storm/src/carl/formula/parser/../Formula.h:599:15: note: 'pbConstraint' has been explicitly marked deprecated here
>             [[deprecated("PBConstraints are normal constraints with Boolean variables now.")]]
>               ^
> 1 warning generated.
> [ 65%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/formula/uninterpreted/UEquality.cpp.o
> [ 68%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/formula/uninterpreted/UFInstance.cpp.o
> [ 72%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/formula/uninterpreted/UFInstanceManager.cpp.o
> [ 72%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/formula/uninterpreted/UFManager.cpp.o
> [ 75%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/formula/uninterpreted/UninterpretedFunction.cpp.o
> [ 75%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/groebner/gb-buchberger/BuchbergerStats.cpp.o
> [ 79%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/numbers/adaption_cln/operations.cpp.o
> [ 79%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/numbers/adaption_float/mpfr_float.cpp.o
> [ 82%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/numbers/adaption_gmpxx/operations.cpp.o
> [ 86%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/numbers/number/NumberClI.cpp.o
> [ 86%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/numbers/number/NumberClRA.cpp.o
> [ 89%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/numbers/number/NumberMpq.cpp.o
> [ 89%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/numbers/number/NumberMpz.cpp.o
> [ 93%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/util/BitVector.cpp.o
> [ 93%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/util/CMakeOptions.cpp.o
> [ 96%] Building CXX object src/carl/CMakeFiles/lib_carl.dir/util/debug.cpp.o
> [100%] Linking CXX shared library ../../libcarl.dylib
> [100%] Built target lib_carl_
> 

but when I try to install storm, there is error when I do `cmake ..`, which says lib_carl does not exit
```
CMake Deprecation Warning at CMakeLists.txt:3 (cmake_policy):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- The CXX compiler identification is AppleClang 14.0.3.14030022
-- The C compiler identification is AppleClang 14.0.3.14030022
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Storm - CMake install dir: /usr/local/lib/CMake/storm
-- Storm - Building RELEASE version.
-- Storm - Could not find ccache.
-- Storm - Detected operating system Mac OS.
-- Storm - Detected that target system uses Apple Silicon.
CMake Warning at CMakeLists.txt:194 (message):
  Compiling natively on Apple Silicon is experimental.  Please report issues
  to support@stormchecker.org.  For more information visit
  https://www.stormchecker.org/documentation/obtain-storm/apple-silicon.html


CMake Warning at CMakeLists.txt:197 (message):
  CLN and GiNaC are currently not supported on Apple Silicon-based
  architectures.  Disabling Storm and carl usage of the libraries.


-- Storm - Assuming extension for shared libraries: .dylib
-- Storm - Assuming extension for static libraries: .a
-- Storm - Build static libraries.
CMake Warning at CMakeLists.txt:247 (message):
  Disabling stack checks for AppleClang version 11.0.0 or higher.


-- Storm - Enabling link-time optimizations.
-- Performing Test COMPILER_C_WORKS
-- Performing Test COMPILER_C_WORKS - Success
-- Performing Test COMPILER_CXX_WORKS
-- Performing Test COMPILER_CXX_WORKS - Success
-- Storm - Using compiler configuration AppleClang 14.0.3.14030022.
-- Storm - Building external resources with 5 job(s) in parallel.
-- Storm - Including Eigen 3.4.0 commit b0eded878d5d162d61583a286c0d8a45406ad1bc.
CMake Warning (dev) at resources/3rdparty/CMakeLists.txt:103 (find_package):
  Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
  --help-policy CMP0167" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

Call Stack (most recent call first):
  CMakeLists.txt:473 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Storm - Using workaround for Boost >= 1.81
-- Storm - Using boost 108500 (library version 1_85).
-- Storm - Including ExprTk.
-- Storm - Including Parallel Hashmap.
-- Storm - Including cpphoafparser 0.99.2
-- Storm - Including ModernJSON.
-- Storm - Linking with Z3 (version 4.13.0). (Z3 version supports optimization)
-- Storm - Using system version of glpk.
-- Storm - Linking with glpk 5.0
CMake Warning at resources/3rdparty/include_cudd.cmake:39 (message):
  There are some known issues compiling CUDD on some setups.  We implemented
  a workaround that mostly works, but if you still have problems compiling
  CUDD, especially if you do not use the default compiler of your system,
  please contact the Storm developers.
Call Stack (most recent call first):
  resources/3rdparty/CMakeLists.txt:287 (include)
  CMakeLists.txt:473 (include)


-- Storm - Linking with CUDD 3.0.0.
CMake Error at resources/3rdparty/CMakeLists.txt:310 (get_target_property):
  get_target_property() called with non-existent target "lib_carl".
Call Stack (most recent call first):
  CMakeLists.txt:473 (include)


CMake Error at resources/3rdparty/CMakeLists.txt:325 (message):
  File does not exist, did you build carl?
Call Stack (most recent call first):
  CMakeLists.txt:473 (include)


-- Configuring incomplete, errors occurred!
```

**Comment by volkm** [comment]
You mentioned in another issue (https://github.com/moves-rwth/storm/issues/583) that pycarl was successfully built. Is this issue still open?
If yes, can you provide the full output of the build step `python setup.py ...`?

**Comment by xwang2775** [comment]
Thanks so much for your help! 

I somehow managed to install pycarl using `pip install -ve .`, but there still has same problems when using `python3 setup.py develop`.

I'm not sure if that matters, I'll try installing the latest version of storm and see how it looks.

**Comment by xwang2775** [comment]
I'm sorry but I think pycarl still failed to install, the out put of `python3 setup.py develop` looks like this

```
(env) (/Users/wangxinyu/RAG/RAG-with-Verification/rag) wangxinyu@wangxinyu pycarl % python3 setup.py develop
/Users/wangxinyu/RAG/RAG-with-Verification/rag/lib/python3.11/site-packages/setuptools/__init__.py:85: _DeprecatedInstaller: setuptools.installer and fetch_build_eggs are deprecated.
!!

        ********************************************************************************
        Requirements should be satisfied by a PEP 517 installer.
        If you are using pip, you can try `pip install --use-pep517`.
        ********************************************************************************

!!
  dist.fetch_build_eggs(dist.setup_requires)
running develop
/Users/wangxinyu/RAG/RAG-with-Verification/rag/lib/python3.11/site-packages/setuptools/command/develop.py:42: EasyInstallDeprecationWarning: easy_install command is deprecated.
!!

        ********************************************************************************
        Please avoid running ``setup.py`` and ``easy_install``.
        Instead, use pypa/build, pypa/installer or other
        standards-based tools.

        See https://github.com/pypa/setuptools/issues/917 for details.
        ********************************************************************************

!!
  easy_install.initialize_options(self)
/Users/wangxinyu/RAG/RAG-with-Verification/rag/lib/python3.11/site-packages/setuptools/_distutils/cmd.py:66: SetuptoolsDeprecationWarning: setup.py install is deprecated.
!!

        ********************************************************************************
        Please avoid running ``setup.py`` directly.
        Instead, use pypa/build, pypa/installer or other
        standards-based tools.

        See https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html for details.
        ********************************************************************************

!!
  self.initialize_options()
running egg_info
creating lib/pycarl.egg-info
writing lib/pycarl.egg-info/PKG-INFO
writing dependency_links to lib/pycarl.egg-info/dependency_links.txt
writing requirements to lib/pycarl.egg-info/requires.txt
writing top-level names to lib/pycarl.egg-info/top_level.txt
writing manifest file 'lib/pycarl.egg-info/SOURCES.txt'
reading manifest file 'lib/pycarl.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'lib/pycarl.egg-info/SOURCES.txt'
running build_ext
Pycarl - Building into build/temp.macosx-11.0-arm64-cpython-311
Pycarl - We only support carl-storm (https://github.com/moves-rwth/carl-storm) indicated by version 14.x. On this system, we only found version 24.06 at /usr/local/lib/cmake/carl
Pycarl - Using carl 24.06 from /usr/local/lib/cmake/carl
Pycarl - Using pybind11 version 2.10.0
Pycarl - WARNING: No parser support!
Pycarl - WARNING: No support for CLN!
Pycarl - CMake args=['-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=/Users/wangxinyu/RAG/RAG-with-Verification/pycarl/lib/pycarl', '-DPython_EXECUTABLE=/Users/wangxinyu/RAG/RAG-with-Verification/rag/bin/python3', '-DCMAKE_BUILD_TYPE=Release', '-DPYBIND_VERSION=2.10.0', '-DCARL_DIR_HINT=/usr/local/lib/cmake/carl', '-DUSE_PARSER=OFF', '-DUSE_CLN_NUMBERS=OFF']
-- The C compiler identification is AppleClang 14.0.3.14030022
-- The CXX compiler identification is AppleClang 14.0.3.14030022
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Python: /Users/wangxinyu/RAG/RAG-with-Verification/rag/bin/python3 (found version "3.11.9") found components: Interpreter Development Development.Module Development.Embed
-- Pycarl - Using shipped version of pybind11 (version 2.10.0).
CMake Deprecation Warning at build/temp.macosx-11.0-arm64-cpython-311/_deps/pybind11-src/CMakeLists.txt:8 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- pybind11 v2.10.0 
-- Performing Test HAS_FLTO
-- Performing Test HAS_FLTO - Success
-- Performing Test HAS_FLTO_THIN
-- Performing Test HAS_FLTO_THIN - Success
-- Configuring done (15.5s)
-- Generating done (0.0s)
-- Build files have been written to: /Users/wangxinyu/RAG/RAG-with-Verification/pycarl/build/temp.macosx-11.0-arm64-cpython-311
[ 80%] Building CXX object CMakeFiles/core.dir/src/core/monomial.cpp.o
[ 80%] Building CXX object CMakeFiles/core.dir/src/core/variable.cpp.o
[ 80%] Building CXX object CMakeFiles/core.dir/src/core/bound_type.cpp.o
[ 80%] Building CXX object CMakeFiles/core.dir/src/mod_core.cpp.o
[100%] Linking CXX shared module /Users/wangxinyu/RAG/RAG-with-Verification/pycarl/lib/pycarl/core.cpython-311-darwin.so
ld: library not found for -llib_carl
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [/Users/wangxinyu/RAG/RAG-with-Verification/pycarl/lib/pycarl/core.cpython-311-darwin.so] Error 1
make[2]: *** [CMakeFiles/core.dir/all] Error 2
make[1]: *** [CMakeFiles/core.dir/rule] Error 2
make: *** [core] Error 2
Traceback (most recent call last):
  File "/Users/wangxinyu/RAG/RAG-with-Verification/pycarl/setup.py", line 182, in <module>
    setup(
  File "/Users/wangxinyu/RAG/RAG-with-Verification/rag/lib/python3.11/site-packages/setuptools/__init__.py", line 108, in setup
    return distutils.core.setup(**attrs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/wangxinyu/RAG/RAG-with-Verification/rag/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 184, in setup
    return run_commands(dist)
           ^^^^^^^^^^^^^^^^^^
  File "/Users/wangxinyu/RAG/RAG-with-Verification/rag/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 200, in run_commands
    dist.run_commands()
  File "/Users/wangxinyu/RAG/RAG-with-Verification/rag/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 970, in run_commands
    self.run_command(cmd)
  File "/Users/wangxinyu/RAG/RAG-with-Verification/rag/lib/python3.11/site-packages/setuptools/dist.py", line 956, in run_command
    super().run_command(command)
  File "/Users/wangxinyu/RAG/RAG-with-Verification/rag/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 989, in run_command
    cmd_obj.run()
  File "/Users/wangxinyu/RAG/RAG-with-Verification/rag/lib/python3.11/site-packages/setuptools/command/develop.py", line 36, in run
    self.install_for_development()
  File "/Users/wangxinyu/RAG/RAG-with-Verification/rag/lib/python3.11/site-packages/setuptools/command/develop.py", line 113, in install_for_development
    self.run_command('build_ext')
  File "/Users/wangxinyu/RAG/RAG-with-Verification/rag/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
    self.distribution.run_command(command)
  File "/Users/wangxinyu/RAG/RAG-with-Verification/rag/lib/python3.11/site-packages/setuptools/dist.py", line 956, in run_command
    super().run_command(command)
  File "/Users/wangxinyu/RAG/RAG-with-Verification/rag/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 989, in run_command
    cmd_obj.run()
  File "/Users/wangxinyu/RAG/RAG-with-Verification/pycarl/setup.py", line 154, in run
    subprocess.check_call(["cmake", "--build", ".", "--target", ext.name] + build_args, cwd=self.build_temp)
  File "/Users/wangxinyu/RAG/RAG-with-Verification/rag/lib/python3.11/subprocess.py", line 413, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '--build', '.', '--target', 'core', '--config', 'Release', '--', '-j8']' returned non-zero exit status 2.

```

**Comment by volkm** [comment]
The carl version in `/usr/local/lib/cmake/carl` is definitely too old. My guess it that it was either installed via homebrew at some point (and can be uninstalled with homebrew) or was manually installed at some (and must be manually deleted). After removing the directory, try to restart the configuration and build process.

**Comment by xwang2775** [comment]
I tried removing carl from the directory` /usr/local/lib/cmake/carl` and it worked! Thanks so much!
